frontend-react/src/components/SenderModeBanner.tsx

```diff
@@ -1,23 +1,83 @@
-import { ReactElement } from "react";
+import { ReactElement, useMemo } from "react";
 import { IconWarning } from "@trussworks/react-uswds";
 import { NavLink } from "react-router-dom";
 
-import useSenderMode from "../hooks/UseSenderMode";
 import { useSessionContext } from "../contexts/SessionContext";
+import useRequestConfig from "../hooks/network/UseRequestConfig";
+import { API, createRequestConfig, Endpoint } from "../network/api/NewApi";
+import { Sender } from "../network/api/OrgApi";
+import { MemberType } from "../hooks/UseOktaMemberships";
 
 const isNotActive = (val: string): boolean => {
     return val === "testing" || val === "inactive";
 };
 
-const SenderModeBanner = (): ReactElement | null => {
-    const session = useSessionContext();
-    const status = useSenderMode(
-        session.store.values.org,
-        session.store.values.senderName
+// Class is useless right now, no run-time type check
+class RSSender {}
+const SenderAPI: API = {
+    resource: RSSender,
+    baseUrl: "/api/settings/organizations",
+    endpoints: new Map<string, Endpoint>([
+        [
+            "list",
+            {
+                url: "/:org/senders",
+                methods: ["GET"],
+            },
+        ],
+        [
+            "detail",
+            {
+                url: "/:org/senders/:sender",
+                methods: ["GET"],
+            },
+        ],
+    ]),
+};
+
+const useSenderResource = () => {
+    /* Access the session. */
+    const { memberships, oktaToken } = useSessionContext();
+    /* Create a stable config reference with useMemo(). */
+    const config = useMemo(
+        () =>
+            createRequestConfig<{ org: string; sender: string }>(
+                SenderAPI,
+                "detail",
+                "GET",
+                oktaToken?.accessToken,
```

Comment:
[thoughts towards a future refactor] I'm still thinking about a way that we could simplify the signature of this config, and one thing that's now occurring to me when I see this in action is that for a given application state, all request configs will have the same access token and organization. With that in mind there is likely a way to abstract away the need the pass in those parameters to each hook, and instead have the config creation process itself be informed about the application state. Maybe it's a matter of providing a request config creation function via context the same way we're providing the raw authenticated user info now?

Reply:
Your thought on the last PR about `useRequestConfig` is funny because now, it seems like a hook-ified way to create the request config is necessary and that'd be the ideal name for it now that it's needed 😂

Reply:
Basically, what `useSenderResource` is doing in this PR -- accessing the session and passing those vars in -- should be handled by a custom hook producing the config and extracted from everything else. THAT is almost what I was hoping `useEndpoint` would do; take as few params as possible and produce a config...but then also pass that config in and get data. I think this is a better approach. I like the idea of there being a hook to create a stable config that runs solely on those bare bones params we want to input.